src/main/java/application/web/MenuController.java

```diff
@@ -0,0 +1,45 @@
+package application.web;
+
+import application.model.Client;
+import application.model.Dish;
+import application.services.DefaultMenuService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.SessionAttributes;
+import org.springframework.web.servlet.ModelAndView;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Created by MSI on 09.09.2016.
+ */
+@Controller
+@SessionAttributes(types = Client.class)
+public class MenuController {
+
+    @Autowired
+    DefaultMenuService service;
+
+    @RequestMapping("/showMenu")
+    public ModelAndView requestHandlingMethod(HttpServletRequest request, @ModelAttribute Client client) {
+        Optional<String> sorter = null;
+        sorter = Optional.ofNullable(request.getParameter("sorter"));
+
+        List<Dish> menu;
```

Comment:
switch to java 8

List<Dish> menu = sorter.map( s -> service.performSortedSearch(sorter.get())).orElse(menu = service.getMenu());